src/Umbraco.Web.UI.Client/src/packages/core/temporary-file/temporary-file-manager.class.ts

```diff
@@ -71,9 +77,58 @@ export class UmbTemporaryFileManager<
 		return filesCompleted;
 	}
 
+	async #validateItem(item: UploadableItem): Promise<boolean> {
+		const maxFileSize = await this.observe(this.#temporaryFileConfigRepository.part('maxFileSize')).asPromise();
+		if (maxFileSize && item.file.size > maxFileSize) {
+			const notification = await this.getContext(UMB_NOTIFICATION_CONTEXT);
+			notification.peek('warning', {
+				data: {
+					headline: 'Upload',
+					message: `
+${this.#localization.term('media_invalidFileSize')}: ${item.file.name} (${formatBytes(item.file.size)}).
+
+${this.#localization.term('media_maxFileSize')} ${formatBytes(maxFileSize)}.
+					`,
+					whitespace: 'pre-line',
+				},
+			});
+			return false;
+		}
+
+		const fileExtension = item.file.name.split('.').pop() ?? '';
+
+		const [allowedExtensions, disallowedExtensions] = await this.observe(
+			observeMultiple([
+				this.#temporaryFileConfigRepository.part('allowedUploadedFileExtensions'),
+				this.#temporaryFileConfigRepository.part('disallowedUploadedFilesExtensions'),
+			]),
+		).asPromise();
+
+		if (
+			(allowedExtensions && !allowedExtensions.includes(fileExtension)) ||
```

Comment:
On my initial test, I didn't make any changes to my configuration, it has `allowedUploadedFileExtensions: []`. I was unable to upload any media (of any extension).
Should it be checking that `allowedExtensions` has any values? e.g. `allowedExtensions?.length`.

Reply:
Oh yes, of course. Good catch!

I also just noticed the max file size from the server is in kilobytes, so we need to multiply by 1024 before validating.